src/components/Contacts/ContactsTable/ContactsTable.tsx

```diff
@@ -84,14 +84,42 @@ export const ContactsTable: React.FC<Props> = ({
             renderEmpty()
           ) : (
             <div data-testid="ContactRows">
-              {data.contacts.nodes?.map((contact) => (
-                <ContactRow
-                  accountListId={accountListId}
-                  key={contact.id}
-                  contact={contact}
-                  onContactSelected={handleOnContactSelected}
-                />
-              ))}
+              <Virtuoso
+                data={data.contacts.nodes}
+                style={{ height: 'calc(100vh - 160px)' }}
+                endReached={() =>
+                  fetchMore({
+                    variables: { after: data.contacts.pageInfo.endCursor },
+                    updateQuery: (prev, { fetchMoreResult }) => {
+                      if (!fetchMoreResult) {
+                        return prev;
+                      }
+                      return {
+                        ...prev,
+                        ...fetchMoreResult,
+                        contacts: {
+                          ...prev.contacts,
+                          ...fetchMoreResult.contacts,
+                          pageInfo: fetchMoreResult.contacts.pageInfo,
```

Comment:
Is this line needed? Seems like the previous line should have just set it.

Reply:
Good point, I'll remove this line.